HPCC-21258 Track CallerId through HPCC components

afishbeck opened the pull request:
Also support "Global-Id", "Caller-Id" versions of the id tracking
HTTP headers.  Also remember the form of HTTP header used and use
the same in SOAPCALL and HTTPCALL.

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

HPCCSmoketest commented:
Process of PR-12310, label: trackCallerIdForECL starts now.
The reason of this test is: Did not build yet.
Commit ID: c42d5fcbbdf181423446b1bfd1713d1984bf5895
Estimated completion time is ~0.5 hour(s)

afishbeck commented on rtl/include/eclhelper.hpp:
```diff
@@ -44,8 +44,8 @@ typedef unsigned short UChar;
 //that a work unit can't be rerun.  Try as hard as possible to retain compatibility.
-#define ACTIVITY_INTERFACE_VERSION      650
-#define MIN_ACTIVITY_INTERFACE_VERSION  650             //minimum value that is compatible with current interface
+#define ACTIVITY_INTERFACE_VERSION      651
```
@ghalliday this seems like overkill when the only way ECL can access the new function is through a new plugin method.  But a workunit that uses the new plugin feature will not be backward compatible with 7.0.x.

Is there a better way?

richardkchapman replied:
I don't think we change the interface version for cases where new functions have been added and only new code uses it.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: c42d5fcbbdf181423446b1bfd1713d1984bf5895
Build: success
Build: success
Install HPCC Platform
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |104 |104 |0 |0 |0 |32 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |25 sec  (00:00:25) |
|setup (thor) |11 |11 |0 |43 sec  (00:00:43) |
|setup (roxie) |11 |11 |0 |17 sec  (00:00:17) |
|test (hthor) |820 |820 |0 |248 sec  (00:04:08) |
|test (thor) |745 |745 |0 |586 sec  (00:09:46) |
|test (roxie) |894 |894 |0 |349 sec  (00:05:49) |


HPCC Stop: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   27 sec (00:00:27) |  191 sec (00:03:11) |    0 sec (00:00:00) |    3 sec (00:00:03) |   18 sec (00:00:18) | 1464 sec (00:24:24) |   20 sec (00:00:20) | 1723 sec (00:28:43) |

afishbeck commented:
@richardkchapman removed eclhelper version number increment.

HPCCSmoketest commented:
This PR (12310) is in the 1/2 position of the queue. Estimated start time is after ~0.5 hour(s)

richardkchapman commented:
@wangkx Please review

HPCCSmoketest commented:
Process of PR-12310, label: trackCallerIdForECL starts now.
The reason of this test is: New commit.
Commit ID: 22eb9f601d177a7f3a9a252316785d8ec1fd2f16
Estimated completion time is ~0.5 hour(s)

wangkx commented on thorlcr/graph/thgraph.cpp:
```diff
@@ -2577,6 +2582,10 @@ class CThorContextLogger : implements IContextLogger, public CSimpleInterface
     }
+    virtual const char *queryCallerId() const override
+    {
+        return localId.str();
```
Should be callerId?

afishbeck replied:
Yes, good catch, thanks.

wangkx reviewed (commented):
@afishbeck one question.

And can someone review the globalid.ecl?

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 22eb9f601d177a7f3a9a252316785d8ec1fd2f16
Build: success
Build: success
Install HPCC Platform
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |104 |104 |0 |0 |0 |34 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |7 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |24 sec  (00:00:24) |
|setup (thor) |11 |11 |0 |43 sec  (00:00:43) |
|setup (roxie) |11 |11 |0 |18 sec  (00:00:18) |
|test (hthor) |820 |820 |0 |249 sec  (00:04:09) |
|test (thor) |745 |745 |0 |596 sec  (00:09:56) |
|test (roxie) |894 |894 |0 |347 sec  (00:05:47) |


HPCC Stop: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   32 sec (00:00:32) |  195 sec (00:03:15) |    0 sec (00:00:00) |    2 sec (00:00:02) |   18 sec (00:00:18) | 1472 sec (00:24:32) |   20 sec (00:00:20) | 1739 sec (00:28:59) |

afishbeck committed:
HPCC-21258 Track CallerId through HPCC components

Also support "Global-Id", "Caller-Id" versions of the id tracking
HTTP headers.  Also remember the form of HTTP header used and use
the same in SOAPCALL and HTTPCALL.

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

afishbeck commented:
@wangkx updated, please check.

wangkx commented:
@afishbeck looks good.
